src/main/scala/com/codacy/duplication/pmd/Cpd.scala

```diff
@@ -98,31 +109,38 @@ object Cpd extends DuplicationTool {
           ScalaTokenizer,
           cpdScala.getExtensions.asScala: _*) {}
         Some(cpdConfiguration(scalaLanguage, 50, options))
-      case Languages.CSharp => Some(cpdConfiguration(new CsLanguage, 50, options))
-      case _                => None
+      case _ => None
     }
   }
 
-  private def cpdConfiguration(cpdLanguage: CPDLanguage,
-                               defaultMinToken: Int,
-                               options: Map[DuplicationConfiguration.Key, DuplicationConfiguration.Value]) = {
+  // TODO: Move to codacy-plugins-api
+  implicit def boolean(value: JsValue): Option[Boolean] =
+    Option(value: JsValue).collect { case JsBoolean(bool) => bool }
 
-    val ignoreAnnotations = true
-    val skipLexicalErrors = true
+  implicit def int(value: JsValue): Option[Int] =
+    Option(value: JsValue).collect { case JsNumber(bigDecimal) => bigDecimal.toInt }
 
-    val minTokenMatch: Int = options.get(DuplicationConfiguration.Key("minTokenMatch")).fold(defaultMinToken) {
-      value: DuplicationConfiguration.Value =>
-        Option(value: JsValue).collect {
-          case JsNumber(number) => number.toInt
-          case _                => defaultMinToken
-        }.getOrElse(defaultMinToken)
+  implicit class DuplicationConfigurationExtended(
+    options: Map[DuplicationConfiguration.Key, DuplicationConfiguration.Value]) {
+
+    def getValue[A](key: DuplicationConfiguration.Key, defaultValue: A)(implicit ev: JsValue => Option[A]): A = {
+      options.get(key).fold(defaultValue) { value: DuplicationConfiguration.Value =>
+        Option(value: JsValue).flatMap(ev).getOrElse(defaultValue)
+      }
     }
+  }
 
+  private def cpdConfiguration(cpdLanguage: CPDLanguage,
+                               defaultMinToken: Int,
+                               options: Map[DuplicationConfiguration.Key, DuplicationConfiguration.Value]) = {
     val cfg = new CPDConfiguration()
     cfg.setLanguage(cpdLanguage)
-    cfg.setIgnoreAnnotations(ignoreAnnotations)
-    cfg.setSkipLexicalErrors(skipLexicalErrors)
-    cfg.setMinimumTileSize(minTokenMatch)
+    cfg.setIgnoreAnnotations(options.getValue[Boolean](DuplicationConfiguration.Key("ignoreAnnotations"), true))
```

Comment:
This seems a bit too verbose and does not say what it is actually doing.
* I would prefer`getOrElse` instead of `getValue` to make the else explicit
* Since you are not using the `DuplicationConfiguration.Key` for anything why not pass just a string? Or Maybe have all the options somewhere and just use them by name here.
* You should be able to infer the type from the second argument and not need to make it explicit in the first place

Reply:
Not sure if it's possible. I can't override any method name that already exists on `Map` like `get` or `getOrElse`.

Reply:
What about the other two points?